moveit_core/planning_scene/src/pyplanning_scene.cpp

```diff
@@ -0,0 +1,96 @@
+
+auto GET_TRANSFORMS1 = py::overload_cast<>(&PlanningScene::getTransforms, py::const_);
+
+auto IS_STATE_VALID1 = py::overload_cast<const moveit_msgs::RobotState&,  //
+                                         const std::string&,              //
+                                         bool>(&PlanningScene::isStateValid, py::const_);
+auto IS_STATE_VALID2 = py::overload_cast<const moveit::core::RobotState&,  //
+                                         const std::string&,               //
+                                         bool>(&PlanningScene::isStateValid, py::const_);
+auto IS_STATE_VALID3 = py::overload_cast<const moveit_msgs::RobotState&,   //
+                                         const moveit_msgs::Constraints&,  //
+                                         const std::string&,               //
+                                         bool>(&PlanningScene::isStateValid, py::const_);
+auto IS_STATE_VALID4 = py::overload_cast<const moveit::core::RobotState&,  //
+                                         const moveit_msgs::Constraints&,  //
+                                         const std::string&,               //
+                                         bool>(&PlanningScene::isStateValid, py::const_);
+auto IS_STATE_VALID5 = py::overload_cast<const moveit::core::RobotState&,                       //
+                                         const kinematic_constraints::KinematicConstraintSet&,  //
+                                         const std::string&,                                    //
+                                         bool>(&PlanningScene::isStateValid, py::const_);
+
+auto SET_CURRENT_STATE1 = py::overload_cast<const moveit_msgs::RobotState&>(&PlanningScene::setCurrentState);
+auto SET_CURRENT_STATE2 = py::overload_cast<const robot_state::RobotState&>(&PlanningScene::setCurrentState);
+
+void def_planning_scene_bindings(py::module& m)
+{
+  m.doc() = "The planning scene represents the state of the world and the robot, "
+            "and can be used for collision checking";
+  py::class_<PlanningScene, PlanningScenePtr>(m, "PlanningScene")
+      .def(py::init<const moveit::core::RobotModelConstPtr&, const collision_detection::WorldPtr&>(),
+           py::arg("robot_model"), py::arg("world") = collision_detection::WorldPtr(new collision_detection::World()))
+      .def("checkSelfCollision", CHECK_SELF_COLLISION1)
+      .def("checkSelfCollision", CHECK_SELF_COLLISION2)
+      .def("checkCollision", CHECK_COLLISION1)
+      .def("checkCollision", CHECK_COLLISION2)
+      .def("getCurrentStateNonConst", &PlanningScene::getCurrentStateNonConst)
+      .def("getCurrentState", &PlanningScene::getCurrentState)
+      .def("getAllowedCollisionMatrix", &PlanningScene::getAllowedCollisionMatrix)
+      .def("isStateConstrained", IS_STATE_CONSTRAINED1, py::arg("state"), py::arg("constr"), py::arg("verbose") = false)
```

Comment:
If you really need all those method variants, here, I do see the readability improvement of using previously declared method overloads.